src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -281,12 +285,11 @@ public DueDateStat countNumberOfFinishedAssessmentsForExercise(Long exerciseId,
      * @param correctionRounds - the correction round we want finished assessments for
      * @return an array of the number of assessments for the exercise for a given correction round
      */
-    public DueDateStat[] countNumberOfFinishedAssessmentsForExerciseByCorrectionRound(Exercise exercise, int correctionRounds) {
+    public DueDateStat[] countNumberOfFinishedAssessmentsForExerciseForCorrectionRound(Exercise exercise, int correctionRounds) {
```

Comment:
can you move this method into the ResultRepository (as default method) so that the `ExerciseService` does not depend on the `ResultService`?

Please do that for similar count methods as well

Reply:
Done, I moved all count methods to the respective repository